Asset/documentdb-dotnet-todo-app/src/DocumentDBRepository.cs

```diff
@@ -79,7 +79,7 @@ public static async Task DeleteItemAsync(string id, string category)
             await client.DeleteDocumentAsync(UriFactory.CreateDocumentUri(DatabaseId, CollectionId, id), new RequestOptions(){PartitionKey = new PartitionKey(category)});
         }
 
-        public static void Initialize()
+        public static async Task InitializeAsync()
```

Comment:
You need to `await` any sync methods. In fact, this won't even compile since you used the `async` keyword without any `await`. I recommended doing that because it's generally faster and won't run into deadlocks (at least, not for sync-over-async reasons). But then you also need to get rid of all the `GetAwaiter()` and `Wait()` calls on lines 87, 88, 90, and 91 and use `await` instead.

Reply:
If you want to keep using sync APIs in your app, at least just use `SecretClient.GetSecret()` not `SecretClient.GetSecretAsync()`. Using sync is better, but requires more than what you changed here and usually requires changes throughout the whole program e.g., new ASP.NET models are generally async-only from initialization through request processing and middleware.

Reply:
@heaths - Updated.